contributors/design-proposals/server-get.md

```diff
@@ -0,0 +1,170 @@
+
+Example request:
+
+```
+$ curl https://localhost:8443/api/v1/pods -H "Accept: application/json+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    "headers": [
+        {"name": "Name", "type": "string", "description": "The name of the pod, must be unique ..."},
+        {"name": "Status", "type": "string", "description": "Describes the current state of the pod"},
+        ...
+    ],
+    "items": [
+        {"cells": ["pod1", "Failed - unable to start", ...]},
+        {"cells": ["pod2", "Init 0/2", ...]},
+        ...
+    ]
+}
+```
+
+This representation is also possible to return from a watch:
+
+```
+$ curl https://localhost:8443/api/v1/pods?watch=1 -H "Accept: application/json+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    // headers are printed first, in case the watch holds
+    "headers": [
+        {"name": "Name", "type": "string", "description": "The name of the pod, must be unique ..."},
+        {"name": "Status", "type": "string", "description": "Describes the current state of the pod"},
+        ...
+    ]
+}
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    // headers are not returned here
+    "items": [
+        {"cells": ["pod1", "Failed - unable to start", ...]},
+        ...
+    ]
+}
+```
+
+It can also be returned in CSV form:
+
+```
+$ curl https://localhost:8443/api/v1/pods -H "Accept: text/csv+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+Name,Status,...
+pod1,"Failed - unable to start",...
+pod2,"Init 0/2",...
+...
+```
+
+To support "wide" format, columns may be marked with an optional priority field of increasing integers (default
+priority 0):
+
+```
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    "headers": [
+        ...
+        {"name": "Node Name", "type": "string", "description": "The node the pod is scheduled on, empty if the pod is not yet scheduled", "priority": 1},
+        ...
+    ],
+    ...
+}
+```
+
+To allow label columns, and to enable integrators to build effective UIs, each row may contain object metadata:
+
+```
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    "items": [
+        ...
+        {"cells": [...], "metadata": {"name": "pod1", "namespace": "pod2", "labels": {"a": "1"}, ...}},
+        ...
+    ]
+}
+```
+
+Not all metadata may be ideal to return, so we may wish to identify a minimal set to expose (name, namespace, uid,
+resource version, labels, annotations?).
+
+All fields on the server side are candidates for translation and localization changes can be delivered more
+quickly and to all clients.
+
+Clients that wish to print in an advanced manner may continue to fetch the normal resources.
+
+Third-party resources can more easily implement `get` in this fashion - instead of web dashboards and
```

Comment:
Expectation would be:

On the server side, we would implement TPR printing by looking at the swagger spec or the TPR definition.  The client wouldn't have to have a JSONPath library, they'd just get fields.  If we wanted to implement a webhook for that, we could, and it would be transparent to clients (shameless plug to distract brendan)